packages/terra-infinite-list/src/InfiniteList.jsx

```diff
@@ -0,0 +1,435 @@
+        this.itemsByIndex[index].height = newHeight;
+        updatedHeight = true;
+      }
+      if (!this.itemsByIndex[index].offsetTop || Math.abs(this.itemsByIndex[index].offsetTop - node.offsetTop) > 1) {
+        this.itemsByIndex[index].offsetTop = node.offsetTop;
+      }
+      if (this.itemsByIndex.length === this.childCount) {
+        if (!this.scrollGroups.length) {
+          this.updateScrollGroups();
+        } else if (updatedHeight) {
+          this.adjustHeight();
+        }
+      }
+    }
+  }
+
+  adjustHeight() {
+    if (this.contentNode) {
+      this.itemsByIndex.forEach((item, itemIndex) => {
+        const scrollItemNode = this.contentNode.querySelector(`[data-infinite-list-index="${itemIndex}"]`);
+        if (scrollItemNode) {
+          const newHeight = scrollItemNode.getBoundingClientRect().height;
+          if (!this.itemsByIndex[itemIndex].height || Math.abs(newHeight - this.itemsByIndex[itemIndex].height) > 1) {
+            this.itemsByIndex[itemIndex].height = newHeight;
+          }
+          if (!this.itemsByIndex[itemIndex].offsetTop || Math.abs(this.itemsByIndex[itemIndex].offsetTop - scrollItemNode.offsetTop) > 1) {
+            this.itemsByIndex[itemIndex].offsetTop = scrollItemNode.offsetTop;
+          }
+          this.adjustTrailingItems(itemIndex);
+        }
+      });
+
+      // needs to update offset tops of every other save
+      this.updateScrollGroups();
+      this.boundary = {
+        topBoundryIndex: -1,
+        hiddenTopHeight: -1,
+        bottomBoundryIndex: -1,
+        hiddenBottomHeight: -1,
+      };
+      this.update(null, true);
+    }
+  }
+
+  adjustTrailingItems(index) {
+    let lastTop = this.itemsByIndex[index].offsetTop;
+    let lastHeight = this.itemsByIndex[index].height;
+    for (let i = index + 1; i < this.itemsByIndex.length; i += 1) {
+      lastTop += lastHeight;
+      lastHeight = this.itemsByIndex[i].height;
+      this.itemsByIndex[i].offsetTop = lastTop;
+    }
+  }
+
+  wrapChild(child, index) {
+    const wrappedCallBack = (node) => {
+      this.updateHeight(node, index);
+      if (child.props.refCallback) {
+        child.props.refCallback(node);
+      }
+    };
+
+    let newProps = {};
+    if (this.props.isSelectable) {
+      const wrappedOnClick = SelectableUtils.wrappedOnClickForItem(child, index, this.props.onChange);
+      const wrappedOnKeyDown = SelectableUtils.wrappedOnKeyDownForItem(child, index, this.props.onChange);
+      newProps = SelectableUtils.newPropsForItem(child, index, wrappedOnClick, wrappedOnKeyDown, this.props.hasChevrons, this.props.selectedIndexes, this.props.disableUnselectedItems);
+    }
+
+    newProps.refCallback = wrappedCallBack;
+    newProps['data-infinite-list-index'] = index;
+    newProps.style = child.props.style ? Object.assign({}, child.props.style, { overflow: 'hidden' }) : { overflow: 'hidden' };
```

Comment:
Do we have jest snapshot test for this, if not we should add one showing we preserve custom inline styles on the cloned children.

Reply:
Updated.